### ISSUE TITLE
Getting discrepancy in the offsets fetched from fetchEarliestOffsets

### ISSUE THREAD

**Issue by rahul-sr** [SOHU-Co/kafka-node#934]
## Bug Report
Getting discrepancy in the offsets fetched from console client "kafka-consumer-groups" and the fetchEarliestOffsets.

### Environment
* Node version: v8.6.0
* Kafka-node version: 2.5.0
* Kafka version: 1.1.0

### 
* Number of Brokers: 1
* Number of partitions for topic: 21

###  Sample Code

```js

let consumer = new kafka.ConsumerGroupStream(options, topic);
let offset = new kafka.Offset(consumer.consumerGroup.client);

offset.fetchEarliestOffsets([topic], function(error, offsets) {
  if (error) {
    console.log(error);
  }
  console.log(offsets);
});
```

### Output of Above code
{
  "test_topic": {
    "0": 264757,
    "1": 258597,
    "2": 259330,
    "3": 257335,
    "4": 257101,
    "5": 258080,
    "6": 256891,
    "7": 255644,
    "8": 256292,
    "9": 257020,
    "10": 256935,
    "11": 256195,
    "12": 257172,
    "13": 256113,
    "14": 256259,
    "15": 256933,
    "16": 255456,
    "17": 255791,
    "18": 254277,
    "19": 255977,
    "20": 254738
  }
};

### kafka-consumer-groups Output
<img width="294" alt="screen shot 2018-04-15 at 4 27 56 pm" src="https://user-images.githubusercontent.com/16071840/38777706-014380aa-40ca-11e8-9a7d-653e36c6a520.png">

### Debug mode output
kafka-node-client fetchOffset Response: {"c8.aq_cr_snapad_prod":{"0":259719,"1":253277,"2":253553,"3":252041,"4":252547,"5":252949,"6":252123,"7":249735,"8":250967,"9":251637,"10":251479,"11":251075,"12":252355,"13":251101,"14":251446,"15":251661,"16":250327,"17":250343,"18":249220,"19":250915,"20":249669}}

### Issue
I realised that output in debug mode of kafka-node shows correct offsets in the line "kafka-node-client fetchOffset Response" as given above, that is consistent with the "kafka-consumer-groups" output.
However, I want to retreive the same offsets through the APIs of ConsumerGroupStream/ ConsumerGroup.
I tried offset.fetch using current time, offset.fetchEarliestOffsets, however unable to get the correct "CURRENT-OFFSET" of the partitions.
What should I use to get the "CURRENT-OFFSET"?

Any help would be greatly appreciated!

**Comment by rahul-sr** [comment]
`
let consumer = new kafka.ConsumerGroupStream(options, topic);
consumer.consumerGroup.fetchOffset(payloads, function (error, offsets) {
            if (error) {
                reject(error);
            }
            resolve(offsets);
        });
`
Can we use fetchOffset as in the code above @hyperlink ? It gives correct offsets, however, the API
is not mentioned in the docs. I tried to get it by checking the source-code.

**Comment by hyperlink** [comment]
Yes, you can use ConsumerGroups `fetchOffset` to get the current committed offset for the group.

Also `offset.fetchEarliestOffsets` gives the earliest available retained offsets for the topic/partition which is not the same as committed offset for the group's topic/partition.

**Comment by rahul-sr** [comment]
Thanks @hyperlink. You can close this issue now.
